includes/class-sensei-course-theme.php

```diff
@@ -76,6 +79,93 @@ public function add_feature_flag_inline_script() {
 		}
 	}
 
+	/**
+	 * Use Sensei Theme template if the theme is set for the current page.
+	 *
+	 * @access private
+	 */
+	public function maybe_use_sensei_theme_template() {
+		if ( ! is_single() || ! in_array( get_post_type(), [ 'lesson', 'quiz' ], true ) ) {
+			return;
+		}
+
+		if ( get_post_type() === 'quiz' ) {
+			$lesson_id = Sensei()->quiz->get_lesson_id( get_the_ID() );
+		} else {
+			$lesson_id = get_the_ID();
+		}
+
+		$course_id = Sensei()->lesson->get_course_id( $lesson_id );
+		$theme     = get_post_meta( $course_id, self::THEME_POST_META_NAME, true );
+
+		if ( self::SENSEI_THEME !== $theme ) {
+			return;
+		}
+
+		add_filter( 'sensei_use_sensei_template', '__return_false' );
+		add_filter( 'template_include', [ $this, 'get_wrapper_template' ] );
+		add_filter( 'the_content', [ $this, 'override_template_content' ] );
+		add_filter( 'body_class', [ $this, 'add_sensei_theme_body_class' ] );
+		add_action( 'wp_enqueue_scripts', [ $this, 'enqueue_styles' ] );
+	}
+
+	/**
+	 * Get the wrapper template.
+	 *
+	 * @access private
+	 *
+	 * @return string The wrapper template path.
+	 */
+	public function get_wrapper_template() {
+		return Sensei_Templates::locate_template( 'course-theme/index.php' );
+	}
+
+	/**
+	 * It overrides the template content, loading the respective
+	 * template and rendering the blocks from the template.
+	 *
+	 * @access private
+	 *
+	 * @return string The content with template and rendered blocks.
+	 */
+	public function override_template_content() {
+		// Remove filter to avoid infinite loop.
+		remove_filter( 'the_content', [ $this, 'override_template_content' ] );
+
+		ob_start();
+		Sensei_Templates::get_template( 'course-theme/single-' . get_post_type() . '.php' );
+		$output = ob_get_clean();
+
+		// Return template content with rendered blocks.
+		return do_blocks( $output );
+	}
+
+	/**
+	 * Add Sensei theme body class.
+	 *
+	 * @access private
+	 *
+	 * @param string[] $classes
+	 *
+	 * @return string[] $classes
+	 */
+	public function add_sensei_theme_body_class( $classes ) {
+		$classes[] = 'sensei-course-theme';
+
+		return $classes;
+	}
+
+	/**
+	 * Enqueue styles.
+	 *
+	 * @access private
+	 */
+	public function enqueue_styles() {
+		// phpcs:ignore WordPress.WP.EnqueuedResourceParameters.MissingVersion -- External dependency that should be cached.
+		wp_enqueue_style( 'inter-googlefonts', 'https://fonts.googleapis.com/css?family=Inter:normal,bold' );
```

Comment:
Just leaving a note that we should not include Google fonts without user consent, as it will bring third-party tracking. Probably better to ship the fonts with Sensei.

Reply:
Ah, good point! Updated here: https://github.com/Automattic/sensei/pull/4413/commits/75e91db9999697b220c01e2d74ebae24ab7e0703